bigframes/core/__init__.py

```diff
@@ -229,6 +225,23 @@ def order_by(self, by: Sequence[OrderingExpression]) -> ArrayValue:
     def reversed(self) -> ArrayValue:
         return ArrayValue(nodes.ReversedNode(child=self.node))
 
+    def slice(
+        self, start: Optional[int], stop: Optional[int], step: Optional[int]
+    ) -> ArrayValue:
+        if self.node.order_ambiguous and not (self.session._strictly_ordered):
+            warnings.warn(
+                "Window ordering may be ambiguous, this can cause unstable results.",
```

Comment:
Not really - all slice operations address some aspect of the total ordering, which may not be stable under partial ordering. The only case where it may not be worth warning is a pure reverse slice (df[::-1]) where offsets aren't being addressed.